src/Kerbero.Infrastructure/NukiActions/Repositories/NukiSmartLockExternalRepository.cs

```diff
@@ -0,0 +1,105 @@
+using System.Net;
+using System.Text.Json;
+using FluentResults;
+using Flurl;
+using Flurl.Http;
+using Kerbero.Domain.Common.Errors;
+using Kerbero.Domain.Common.Interfaces;
+using Kerbero.Domain.NukiActions.Models;
+using Kerbero.Domain.NukiActions.Repositories;
+using Kerbero.Domain.NukiAuthentication.Errors.CommonErrors;
+using Kerbero.Domain.NukiAuthentication.Errors.CreateNukiAccountErrors;
+using Kerbero.Domain.NukiAuthentication.Models;
+using Kerbero.Infrastructure.Common.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Kerbero.Infrastructure.NukiActions.Repositories;
+
+public class NukiSmartLockExternalRepository: INukiSmartLockExternalRepository
+{
+	private readonly NukiExternalOptions _options;
+	private readonly ILogger<NukiSmartLockExternalRepository> _logger;
+	private readonly InteractorAsync<NukiAccountAuthenticatedRequestDto, NukiAccountAuthenticatedResponseDto> _authenticationInteractor;
+
+	public NukiSmartLockExternalRepository(IOptions<NukiExternalOptions> options, 
+		InteractorAsync<NukiAccountAuthenticatedRequestDto, NukiAccountAuthenticatedResponseDto> authenticationInteractor, 
+		ILogger<NukiSmartLockExternalRepository> logger)
+	{
+		_options = options.Value;
+		_logger = logger;
+		_authenticationInteractor = authenticationInteractor;
+	}
+	
+	public async Task<Result<NukiSmartLocksListExternalResponseDto>> GetNukiSmartLockList(int nukiAccountId)
+	{
+		try
+		{
+			var result = await AuthenticateToNuki(nukiAccountId);
+			if (result.IsFailed)
+			{
+				return result.ToResult();
```

Comment:
```suggestion
				return result.ToResult();
```
result  to result, non mi sembra molto chiaro

Reply:
Ma result è di tipo Result, è colpa della libreria se devo fare così e non posso tornarlo direttamente...